met_explore/preprocessing.py

```diff
@@ -68,40 +71,49 @@ def add_chebi_ids(self):
-                    # logger.debug("ID and ID_INCHI %s %s" % (chebi_id, chebi_id_inchi))
 
                     # If both exist add the chebi_id (based on the DB identifier) to the DF
                     if chebi_id and chebi_id_inchi:
-                        logger.debug("The ID and chebi_id are both found: %s %s" % (chebi_id, chebi_id_inchi))
-                        self.peak_df.loc[index, 'chebi_id'] = chebi_id
+                        new_chebi_ids.append(chebi_id)
 
                         # Get the chebi inchi if it exists & If it's not null add it to the DF.
-                        inchi = self.chebi_df[self.chebi_df.chebi_id == chebi_id].inchikey.values[0]
-                        logger.debug("The inchi that we found is %s " % inchi)
+                        inchi = temp_dict['inchikey'][chebi_id]
                         if not pd.isnull(inchi):
-                            logger.debug("in the wee setting bit")
-                            self.peak_df.loc[index, 'inchikey'] = inchi
+                            new_inchikeys.append(inchi)
+                        else:
+                            new_inchikeys.append(None)
 
                     elif chebi_id and not chebi_id_inchi:
                         logger.debug('chebi_id and not chebi_id_inchi, adding %s' % chebi_id)
                         self.peak_df.loc[index, 'chebi_id'] = chebi_id
+                        new_chebi_ids.append(chebi_id)
 
                         # Grab the inchi from the chebi_id if it exists.
-                        inchi = self.chebi_df[self.chebi_df.chebi_id == chebi_id].inchikey.values[0]
+                        inchi = temp_dict['inchikey'][chebi_id]
                         if not pd.isnull(inchi):
-                            self.peak_df.loc[index, 'inchikey'] = inchi
+                            new_inchikeys.append(inchi)
+                        else:
+                            new_inchikeys.append(None)
 
                     elif chebi_id_inchi and not chebi_id:
                         logger.debug('chebi_id_inchi and not chebi_id adding chebi_id_inchi %s' % chebi_id_inchi)
                         self.peak_df.loc[index, 'chebi_id'] = chebi_id_inchi
                         # As the chebi_id was obtained using the inchi we can leave that inchi as the original one.
+                        new_chebi_ids.append(chebi_id_inchi)
+                        new_inchikeys.append(inchi)
 
                     elif not chebi_id_inchi and not chebi_id:
                         chebi_id = None
                         logger.debug("No chebi ID found so setting it to None")
-                        self.peak_df.loc[index, 'chebi_id'] = chebi_id
+                        new_chebi_ids.append(chebi_id)
+                        new_inchikeys.append(inchi)
+                else:
+                    new_chebi_ids(set_chebi_id)
+                    new_inchikeys(inchi)
+
```

Comment:
Is this doing anything? should these be append statements?

Reply:
Yeah it's a typo ...